src/include/migraphx/op/nonmaxsuppression.hpp

```diff
@@ -266,31 +283,38 @@ struct nonmaxsuppression
             // iterator to first value of this batch
             auto batch_boxes_start = boxes.begin() + batch_idx * num_boxes * 4;
             auto boxes_heap = filter_boxes_by_score(scores_start, num_boxes, score_threshold);
-            selected_boxes_inside_class.clear();
+            int64_t selected_boxes_inside_class = 0;
             while(not boxes_heap.empty() &&
-                  selected_boxes_inside_class.size() < max_output_boxes_per_class)
+                  selected_boxes_inside_class < max_output_boxes_per_class)
             {
                 // select next top scorer box and remove any boxes from boxes_heap that exceeds IOU
                 // threshold with the selected box
-                const auto next_top_score = boxes_heap.top();
-                boxes_heap.pop();
-                selected_boxes_inside_class.push_back(next_top_score);
+                const auto next_top_score = boxes_heap.front();
+                auto next_box             = batch_box(batch_boxes_start, next_top_score.second);
+                auto next_box_idx         = next_top_score.second;
+                // Poor man's "pop" for vector
+                boxes_heap.erase(boxes_heap.begin());
```

Comment:
The reason why its ugly to write `pop_front` is because its really slow. I think you can just skip the first element in the `copy_if` instead `std::copy_if(std::execution::par, boxes_heap.begin()+1, boxes_heap.end(), ...)`

Reply:
Makes sense. Done